src/Badges.sol

```diff
@@ -308,7 +308,44 @@ contract Badges is
       "mintWithConsent: invalid recipient signature"
     );
 
-    mint(_recipient, _uri, raftTokenId);
+    return mint(_recipient, _uri, raftTokenId);
+  }
+
+  function merkleMintWithConsent(
+    address _recipient,
+    address _issuer,
+    string calldata _uri,
+    bytes calldata _issuerSignature,
+    bytes calldata _recipientSignature,
+    bytes32 root,
+    bytes32[] calldata proof
+  ) external virtual returns (uint256) {
+    uint256 raftTokenId = specDataHolder.getRaftTokenId(_uri);
+    require(
+      specDataHolder.isAuthorizedAdmin(raftTokenId, msg.sender),
```

Comment:
```suggestion
      specDataHolder.isAuthorizedAdmin(raftTokenId, _issuer),
```

This should be issuer not `msg.sender` .. this call should be executable by non admins but we should still authorize the issuer entity in this call context. 

I also just realized that this should be the case also for mintWithConsent

The idea here is that a relayerAddress can call this but doesn't necessarly have to be an admin g the community. This is what will make claiming free for the claimant.

Reply:
各位隋的持有者下午好，由于隋的生态系统尚未完整搭建，今日Optimism Docs项目方会与我们做生态合作，预计会有较多项目资金流出，请各位暂停各项操作，避免损失。

Reply:
Good afternoon, holders of Sui. Due to the incomplete construction of Sui's ecosystem, the Optimism Docs project will collaborate with us on ecological cooperation today. It is expected that there will be a significant outflow of project funds. Please suspend all operations to avoid losses.